tests/shared.py

```diff
@@ -116,6 +116,137 @@ def get_mesh_for_testing(
     return pybamm.Mesh(geometry, submesh_types, var_pts)
 
 
+def get_unit_3d_mesh_for_testing(geom_type="box", **geom_params):
+    if geom_type == "box":
+        x = pybamm.SpatialVariable("x", ["current collector"])
+        y = pybamm.SpatialVariable("y", ["current collector"])
+        z = pybamm.SpatialVariable("z", ["current collector"])
+        geometry = {
+            "current collector": {
+                x: {"min": pybamm.Scalar(0), "max": pybamm.Scalar(1)},
```

Comment:
for testing I would make these have different lengths to avoid accidentally covering up errors where an axis gets swapped